lua/lint.lua

```diff
@@ -305,6 +305,27 @@ local function eval_fn_or_id(x)
   end
 end
 
+--- Run given function with `cwd` set as :cd, restore original cwd afterwards
+---
+---@param cwd string
+---@param fn fun(...):any
+---@return any
+local function with_cwd(cwd, fn, ...)
+  local curcwd = vim.fn.getcwd()
+  if curcwd == cwd then
+    return fn(...)
+  else
+    local mods = { noautocmd = true }
+    vim.cmd.cd({cwd, mods = mods})
+    local ok, result = pcall(fn, ...)
+    vim.cmd.cd({curcwd , mods = mods})
+    if ok then
+      return result
+    end
+    error(result)
+  end
+end
```

Comment:
Just a thought... would it have been better to call `vim.fn.chdir()` instead, which would change directory only for the current window? 🤔